app/models/poll/question.rb

```diff
@@ -69,23 +74,55 @@ def max_votes
     end
   end
 
-  def find_or_initialize_user_answer(user, option_id)
-    option = question_options.find(option_id)
+  def find_or_initialize_user_answer(user, option_id: nil, answer_text: nil)
+    answer = answers.find_or_initialize_by(find_by_attributes(user, option_id))
+
+    if accepts_options?
+      option = question_options.find(option_id)
+      answer.option = option
+      answer.answer = option.title
+    else
+      answer.answer = answer_text
+    end
 
-    answer = answers.find_or_initialize_by(find_by_attributes(user, option))
-    answer.option = option
-    answer.answer = option.title
     answer
   end
 
+  def open_ended_answered_count
+    answers.where.not(answer: [nil, ""]).count
+  end
+
+  def open_ended_unanswered_count
+    valid_participants_count - open_ended_answered_count
+  end
+
+  def open_ended_answered_percentage
+    return 0.0 if open_ended_total_responses.zero?
+
+    (open_ended_answered_count * 100.0) / open_ended_total_responses
+  end
+
+  def open_ended_unanswered_percentage
+    return 0.0 if open_ended_total_responses.zero?
+
+    (open_ended_unanswered_count * 100.0) / open_ended_total_responses
+  end
+
   private
 
-    def find_by_attributes(user, option)
-      case vote_type
-      when "unique", nil
+    def find_by_attributes(user, option_id)
+      if multiple?
+        { author: user, option_id: option_id }
+      else
         { author: user }
-      when "multiple"
-        { author: user, answer: option.title }
       end
     end
+
+    def open_ended_total_responses
+      open_ended_answered_count + open_ended_unanswered_count
+    end
+
+    def valid_participants_count
+      poll.answers.distinct.count(:author_id)
```

Comment:
This wouldn't count voters who have left everything blank :thinking:. So we could write `poll.voters.count`.

Since this method would then become fairly trivial, maybe we could remove it and include `poll.voters.count` in `open_ended_unanswered_count` :thinking:.